src/metabase/query_processor/pivot.clj

```diff
@@ -1,98 +1,144 @@
 (ns metabase.query-processor.pivot
   "Pivot table actions for the query processor"
   (:require [clojure.core.async :as a]
-            [clojure.set :refer [map-invert]]
+            [metabase.mbql.normalize :as normalize]
             [metabase.query-processor :as qp]
             [metabase.query-processor.context :as qp.context]
-            [metabase.query-processor.store :as qp.store]))
+            [metabase.query-processor.context.default :as context.default]
+            [metabase.query-processor.error-type :as qp.error-type]
+            [metabase.query-processor.store :as qp.store]
+            [metabase.util.i18n :refer [tru]]
+            [schema.core :as s]))
 
-(defn powerset
-  "Generate a powerset while maintaining the original ordering as much as possible"
-  [xs]
-  (for [combo (reverse (range (int (Math/pow 2 (count xs)))))]
-    (for [item  (range 0 (count xs))
-          :when (not (zero? (bit-and (bit-shift-left 1 item) combo)))]
-      (nth xs item))))
+(defn- group-number
+  "Come up with a display name given a combination of breakout `indecies` e.g.
 
-(defn- generate-specified-breakouts
-  "Generate breakouts specified by pivot_rows"
```

Comment:
Queries get normalized to lisp-case so I suspect trying to use `snake_case` keys like `pivot_rows` and `pivot_cols` is going to cause problems, so I changed it. We should only use `snake_case` keys for things that directly touch the DB/API boundary e.g. `defendpoint` forms or DB code. Everything else should use `lisp-case`... convert keys in the API endpoints if you need to. Once I get #14113 working this won't be a thing anymore